packages/manager/apps/container/src/container/nav-reshuffle/sidebar/index.tsx

```diff
@@ -191,22 +192,37 @@ const Sidebar = (): JSX.Element => {
     }
   }, [isNavigationSidebarOpened]);
 
+
+  const computeNodeCount = (count: IServicesCount, node: Node): number | boolean => {
+    if (node.count === false) return node.count;
+    return countServices(count, node);
+  };
+
+  const processNode = (count: IServicesCount, node: Node): Node => {
+    return {
+      ...node,
+      count: computeNodeCount(count, node),
+      children: node.children?.map(childNode => processNode(count, childNode))
+    };
+  };
+
   /**
-   * Initialize menu items based on currentNavigationNode
-   */
+ * Initialize menu items based on currentNavigationNode
+ */
   useEffect(() => {
-    const count: ServicesCount = {
-      total: servicesCount?.total,
-      serviceTypes: {
-        ...servicesCount?.serviceTypes,
-      } as Record<string, number>,
+    if (!currentNavigationNode?.children || !servicesCount) return;
```

Comment:
Won't this result in the display of previously displayed menu items when `!currentNavigationNode?.children || !servicesCount` is true ?

It seems to me that we should continue to call setMenuItems with undefined in such case.